src/components/dataset-query/dataset-query.js

```diff
@@ -4,17 +4,35 @@ import './dataset-query.scss'
 import LoadingElement from '../generic-elements/loading-element';
 
 
-const DatasetQuery = ({ defaultQuery, onQueryDataset, queryFailureMessage, isLoading, hasUserSubmittedQuery}) => {
+const DatasetQuery = ({ defaultQuery, onQueryDataset, onCancelQuery, queryFailureMessage, isLoading, hasUserSubmittedQuery }) => {
   const [queryText, setQueryTextRaw] = useState(defaultQuery)
-  const submit = () => { onQueryDataset(queryText) }
+  const [hasUserClickedCancelQuery, setHasUserClickedCancelQuery] = useState(false)
+
+  const submit = () => {
+    onQueryDataset(queryText)
+    setHasUserClickedCancelQuery(false)
+  }
+
+  const cancel = () => {
+    onCancelQuery()
+    setHasUserClickedCancelQuery(true)
+  }
+
   const setQueryText = (e) => setQueryTextRaw(e.target.value)
+  const errorText = hasUserClickedCancelQuery ? 'Your query has been stopped' : 'Query failure.  There may be a syntax issue.'
```

Comment:
Should this be `canceled` instead of `stopped`?

Reply:
We got the verbage from Big Jake. We could track him down and ask him if you like.